realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -230,8 +227,57 @@ static void notifyUserLoggedOut(SyncUser user) {
         }
     }
 
+    // This is called from SyncManager.cpp from the worker thread the Sync Client is running on
+    // Right now Core doesn't send these errors to the proper session, so instead we need to notify all sessions
+    // from here. This can be removed once better error propagation is implemented in Sync Core.
+
+    /**
+     * All errors from native Sync is reported to this method. From the path we can determine which
+     * session to contact. If {@code path == null} all sessions are effected.
+     */
+    @SuppressWarnings("unused")
+    private static synchronized void notifyErrorHandler(int errorCode, String errorMessage, String path) {
+        for (SyncSession syncSession : sessions.values()) {
+            if (path == null || path.equals(syncSession.getConfiguration().getPath())) {
+                syncSession.notifySessionError(errorCode, errorMessage);
+            }
+        }
+    }
+
+    /**
+     * This is called from the Object Store (through JNI) to request an {@code access_token} for
+     * the session specified by sessionPath.
+     *
+     * This will also schedule a timer to proactively refresh the {@code access_token} regularly, before
+     * the {@code access_token} expires.
+     *
+     * @throws IllegalStateException if the wrapped Java session is not found.
+     * @param sessionPath The path to the previously Java wraped session.
+     * @return a valid cached {@code access_token} if available or null.
+     */
+    private synchronized static String bindSessionWithConfig(String sessionPath) {
+        final SyncSession syncSession = sessions.get(sessionPath);
+        if (syncSession == null) {
+            throw new IllegalStateException("Matching Java SyncSession could not be found for: " + sessionPath);
```

Comment:
Will this propagate correctly? I assume it is being called from the SyncClient thread?